packages/libs/eda/src/lib/core/components/computations/plugins/abundance.tsx

```diff
@@ -21,21 +21,24 @@ import { useMemo } from 'react';
 import { ComputationStepContainer } from '../ComputationStepContainer';
 import './Plugins.scss';
 import { makeClassNameHelper } from '@veupathdb/wdk-client/lib/Utils/ComponentUtils';
+import { VariableCollectionSelectList } from '../../variableSelectors/VariableCollectionSelectList';
 
 const cx = makeClassNameHelper('AppStepConfigurationContainer');
 
 export type AbundanceConfig = t.TypeOf<typeof AbundanceConfig>;
 // eslint-disable-next-line @typescript-eslint/no-redeclare
-export const AbundanceConfig = t.type({
+export const AbundanceConfig = t.partial({
   collectionVariable: VariableCollectionDescriptor,
   rankingMethod: t.string,
 });
 
+const CompleteAbundanceConfig = partialToCompleteCodec(AbundanceConfig);
```

Comment:
> are we saying that when we declare some type, it should always reflect possible states rather than the desired state? like is the reason were unhappy w how things were bc a user might simply not have set a rankingMethod yet, even though one is required for the compute?


Essentially, yes. Otherwise we end up with intermediate states that are not reflected by our types. I think we were getting around this with a type cast, and I kept running into runtime errors.

I think this bears more discussion, tbh. There are other ways to approach this, which we should enumerate and evaluate. We should also be consistent with visualizations, in which we don't explicitly declare a "complete" state. Instead we just check that the props we want are defined in the relevant places.

I guess a question to ask is, is this too clever? It also doesn't cover every case (see differential abundance, where we have to drill further into the `comparator` object).